### PR TITLE
Fix toplevel

### PR DESCRIPTION
**Pull request by krisselden**
This supports legacy style component tests and ember islands.

### CODE REVIEW

File: packages/ember-glimmer/lib/renderer.js [hunk 1]
```diff
@@ -122,7 +125,7 @@ class Renderer {
       rootOutletState: UNDEFINED_REFERENCE,
       isTopLevel: true
     });
-    this._renderRoot(view, view.template, self, target, dynamicScope);
+    this._renderRoot(view, RootTemplate.create({ env: this }), self, target, dynamicScope);
```
Comment (krisselden) [inline packages/ember-glimmer/lib/renderer.js]: TODO: cache the root template on the environment

File: packages/ember-glimmer/lib/syntax/curly-component.js [hunk 2]
```diff
@@ -301,6 +301,49 @@ class CurlyComponentManager {
...
+    let component = definition.ComponentClass;
+    dynamicScope.view = component;
+    dynamicScope.targetObject = component;
+
+    component.trigger('didInitAttrs');
+    component.trigger('didReceiveAttrs');
+    component.trigger('willInsertElement');
+    component.trigger('willRender');
+
+    assert(`classNameBindings must not have spaces in them: ${component.toString()}`, (() => {
```
Comment (krisselden) [inline packages/ember-glimmer/lib/syntax/curly-component.js]: DRY up assertions in a runInDebug shared helper function

File: packages/ember-glimmer/lib/helpers/component.js [hunk 3]
```diff
@@ -4,17 +4,10 @@ import { EvaluatedArgs, EvaluatedNamedArgs, EvaluatedPositionalArgs } from 'glim
 import { assert } from 'ember-metal/debug';
 import assign from 'ember-metal/assign';
 
-const CLOSURE_COMPONENT = 'ba564e81-ceda-4475-84a7-1c44f1c42c0e';
+const COMPONENT_DEF_BRAND = 'COMPONENT DEFINITION [id=e59c754e-61eb-4392-8c4a-2c0ac72bfcd4]';
```
Comment (krisselden) [inline packages/ember-glimmer/lib/helpers/component.js]: need to get rid of isClosureComponent and import isComponentDefinition

File: packages/ember-glimmer/tests/unit/template-factory-test.js [hunk 4]
```diff
@@ -55,7 +55,8 @@ moduleFor('Template factory test', class extends RenderingTest {
 
     this.render('{{x-precompiled name="precompiled"}} {{x-compiled name="compiled"}}');
 
-    assert.equal(env._templateCache.misses, 2, 'misses 2');
+    // This extra miss is from rendering the root layout in the test harness component
+    assert.equal(env._templateCache.misses, 3, 'misses 3');
```
Comment (krisselden) [inline packages/ember-glimmer/tests/unit/template-factory-test.js]: this is annoying, the test harness could use component.layoutName instead and register the template

**Comment by rwjblue** [comment]
Just updated.

* Rebased / Squashed / etc
* DRY'ed up component instantiation assertions
* Migrated `appendTo` tests to new infrastructure

**Pull request merged**